src/player_display.cpp

```diff
@@ -763,460 +886,566 @@ Strength - 4;    Dexterity - 4;    Intelligence - 4;    Perception - 4" ) );
-                wrefresh( w_skills );
-
-                werase( w_info );
-
-                if( line < skillslist.size() ) {
-                    fold_and_print( w_info, 0, 1, FULL_SCREEN_WIDTH - 2, c_magenta, selectedSkill->description() );
-                }
-                wrefresh( w_info );
-
-                action = ctxt.handle_input();
-                if( action == "DOWN" ) {
-                    if( size_t( line ) < skillslist.size() - 1 ) {
-                        line++;
-                    }
-                } else if( action == "UP" ) {
-                    if( line > 0 ) {
-                        line--;
-                    }
-                } else if( action == "NEXT_TAB" ) {
-                    werase( w_skills );
-                    mvwprintz( w_skills, 0, 0, c_light_gray, header_spaces.c_str() );
-                    center_print( w_skills, 0, c_light_gray, title_SKILLS );
-                    for( size_t i = 0; i < skillslist.size() && i < size_t( skill_win_size_y ); i++ ) {
-                        const Skill *thisSkill = skillslist[i];
-                        const SkillLevel &level = get_skill_level_object( thisSkill->ident() );
-                        bool can_train = level.can_train();
-                        bool isLearning = level.isTraining();
-                        bool rusting = level.isRusting();
-
-                        nc_color cstatus;
-                        if( rusting ) {
-                            cstatus = isLearning ? c_light_red : c_red;
-                        } else if( !can_train ) {
-                            cstatus = c_white;
-                        } else {
-                            cstatus = isLearning ? c_light_blue : c_blue;
-                        }
-
-                        mvwprintz( w_skills, i + 1,  1, cstatus, "%s:", thisSkill->name().c_str() );
-
-                        if( thisSkill->ident() == skill_id( "dodge" ) ) {
-                            mvwprintz( w_skills, i + 1, 14, cstatus, "%4.1f/%-2d(%2d%%)",
-                                       get_dodge(), level.level(), level.exercise() < 0 ? 0 : level.exercise() );
-                        } else {
-                            mvwprintz( w_skills, i + 1, 19, cstatus, "%-2d(%2d%%)", level.level(),
-                                       ( level.exercise() <  0 ? 0 : level.exercise() ) );
-                        }
-                    }
-                    wrefresh( w_skills );
-                    line = 0;
-                    curtab++;
-                } else if( action == "CONFIRM" ) {
-                    get_skill_level_object( selectedSkill->ident() ).toggleTraining();
-                } else if( action == "QUIT" ) {
-                    done = true;
-                }
         }
     } while( !done );
 
     g->refresh_all();
 }
+
+void player::disp_info()
+{
+    player_display_info display_info( *this );
+    display_info.display( *this );
```

Comment:
Instead of forwarding the player instance to each function of `player_display_info display_info` by storing the reference as member of it.

In fact, it must store the player reference in order to keep consistency. Currently one can create an object with one player instance and later call `display` with a completely different player instance.